docs/tutorial/Advanced/docker.rst

```diff
@@ -0,0 +1,240 @@
+        - it.sh
+        - prune.sh
+        - run.sh
+        - stop.sh
+
+Dockerfile
+++++++++++
+
+``Dockerfile`` is the Docker script used to build the Open3D image.
+
+Tools
++++++
+
+We provide a number of Docker tools for convenience:
+
+- ``attach.sh``
+  Start the Open3D docker container and attach to it using a terminal interface.
+- ``build.sh``
+  Build the Open3D docker image.
+- ``delete.sh``
+  Delete the Open3D image.
+- ``it.sh``
+  Start the Open3D docker container and display container stdout.
+- ``prune.sh``
+  Delete hanging containers and images.
+- ``run.sh``
+  Run the Open3D container. Checkout Open3D and build.
+- ``stop.sh``
+  Stop the Open3D container.
+
+Building the Open3D Docker image will take approximately 10-15 minutes to complete.
+At the end the image will be ~1GB in size.
+
+Running the Open3D Docker container will perform the following steps:
+
+- git clone Open3D master to ``~/Open3D_docker``
+- copy the ``headless_sample.py`` and ``headless_sample.sh`` to ``~/Open3D_docker/build/lib/Tutorial/Advanced``
+- run and detach the Open3D container with the host path ``~/Open3D_docker`` mounted inside the container at ``/root/Open3D``
+- attach a terminal to the Open3D container for command line input from the host side
+
+The Open3D container is automatically removed when stopped.
+None of the Open3D files are removed as they in fact reside on the host due to the Docker bind mounting functionality.
+In order to keep the container around remove the ``-rm`` option in ``it.sh`` and/or ``run.sh``.
+
+Prunning images/containers is useful when modifying/testing a new image.
+
+VNC
+```
+A running Open3D container listens to port 5920 on the host.
+The ``it.sh``, ``run.sh`` and ``attach.sh`` scripts redirect host port 5920 to container port 5900.
+
+This allows remoting into the container using VNC to ``<host ip>:5920``.
+The default password is ``1234`` and can be changed in ``Open3D/issue_17/util/docker/open3d-xvfb/setup/entrypoint.sh``.
+Once connected you can use Open3D as usual.
+
+Headless rendering in terminal
+``````````````````````````````
+
+Sometimes it may be necessary to perform rendering as part of some script automation.
+In order to do this follow the next steps::
+
+$ cd <Open3D path>/utilities/docker/open3d-xvfb/tools
+$ ./build.sh
+$ ./attach.sh
+$ ./headless_sample.sh
+
+The ``headless_sample.sh`` renders some images and saves them to disk.
+The images can be accessed in real time on the host at ``~/Open3D_docker/build/lib/TestData/depth`` and won't go away when the container is stopped/deleted.
+
+Limitations
+```````````
+
+- the ``lxde`` based interface employed in this Docker image needs more configuring.
+  Some things won't work as expected. For example ``lxterminal`` crashes.
+- the resolution is set to 1280x1024x8 when remoting into an Open3D container.
+  Open3D windows are larger than this. The resolution will be increased in the future.
+- the ``headless_sample.py`` sample:
```

Comment:
If head less rendering using OSMesa works, it doesn't necessary to mention here.

Reply:
TODO. good point, need to revisit.